### DISCUSSION TITLE
Load SVG dynamically.

### DISCUSSION THREAD

**Discussion by factorde7** [SVG-Edit/svgedit#522] [Q&A]
My scenario a little bit different. I have to create a rectangle (upon a background image). Then save that svg into my database. Now when I load that editor again, I need to repopulate that rectangle on my canvas once again (if it already exists in DB).

So the point is.. how may I load any SVG programatically? I know there is a button for opening SVG, but I don't want it, I want to gather SVG information from my DB dynamically and then load it programatically.

Any help would be appreciated.

**Comment by iuyiuy** [comment]
There's an svgEditorReady event, where you can load the svg file on load and I've just added info on the save event, where you can hook in to change how it's saved:
https://github.com/SVG-Edit/svgedit/wiki/Events

**Comment by factorde7** [comment]
There is already an extension named "ext-php_savefile.js" in extensions folder. How can I use it when someone clicks "Save Image" button.

```
var clickSave = function() {
                // In the future, more options can be provided here
                var saveOpts = {
                    'images': $.pref('img_save'),
                    'round_digits': 6
                };
                svgCanvas.save(saveOpts);
            };
```

Where in above code, I can invoke php save extension? Please help

**Comment by factorde7** [comment]
I have included this extension in following code as well:

```
defaultExtensions = [
                svg_edit_url+'ext-overview_window.js',
                svg_edit_url+'ext-markers.js',
                svg_edit_url+'ext-connector.js',
                svg_edit_url+'ext-eyedropper.js',
                svg_edit_url+'ext-shapes.js',
                svg_edit_url+'ext-imagelib.js',
                svg_edit_url+'ext-grid.js',
                svg_edit_url+'ext-polygon.js',
                svg_edit_url+'ext-star.js',
                svg_edit_url+'ext-panning.js',
                svg_edit_url+'ext-storage.js',
                svg_edit_url+'ext-php_savefile.js'
            ],
```

**Comment by iuyiuy** [comment]
The save event will be called when clicking the 'save image' button in svg-edit. You will get the data there, which you'll be able to handle as you need.

**Comment by factorde7** [comment]
Thanks for reply, one more question. Is there a way to set initial zoom level to "Fit to canvas"?

**Comment by factorde7** [comment]
ok got it, thanks for quick help :)

**Comment by Dilipkumaryadav** [comment]
sir we want get value  in global variable  not in function
```js
svgEditor.setCustomHandlers({
 save: function(win, data) {
  // Save svg
 }
});
```

like data within save function that means when we want save then script trigger but my need get data out side  function  in a variable
